addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -518,14 +520,39 @@ return AbstractModel.extend({
     _loadColors: function (element, events) {
         if (this.fieldColor) {
             var fieldName = this.fieldColor;
+            var self = this;
             _.each(events, function (event) {
                 var value = event.record[fieldName];
-                event.color_index = _.isArray(value) ? value[0] : value;
+                var color = self.loadParams.filters[fieldName] && _.map(self.loadParams.filters[fieldName].filters, f => f.value) || event.record[fieldName];
+                let allFilter = self.loadParams.filters[fieldName] && _.find(self.loadParams.filters[fieldName].filters, f => f.value === "all") || false;
+                if (allFilter && !allFilter.active) {
+                    color = [event.attendee_id]
+                } else if (allFilter && allFilter.active) {
+                    color = value.includes(self.getSession().partner_id) ? self.getSession().partner_id : value
+                }
+                event.color_index = _.isArray(color) ? self._getColorIndex(value, color[0]) : color;
             });
             this.model_color = this.fields[fieldName].relation || element.model;
         }
         return Promise.resolve();
     },
+    /**
+     * Get the color index used to render the event and the filter.
+     * Since we have a maximum of 30 colors, we have to adapt the color
+     * index in case the id exceeds the number of colors available.
+     * 
+     * @private
+     * @param {Array} list
+     * @param {Integer} color
+     * @returns {Integer}
+     */
+    _getColorIndex: function(list, color) {
+        let new_value = color % 30;
```

Comment:
I know let is better but as the rest of the file uses var, I would better keep the 'var' everywhere
and rename to new_color_index
and rename params + give description in docstring of each param.
list : list of what ?
color: rename to color_index ?